From 458d669a2893d1aa4028a8ce9a44cdabfe3ec6e9 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Tue, 6 Mar 2018 19:03:07 -0500
Subject: [PATCH] Use unique repo name for tests that require credentials.

Otherwise, when running tests in parallel, they might conflict.
Fixes #2899.

Patch-Source: https://github.com/git-lfs/git-lfs/pull/2901
---
 test/cmd/lfstest-gitserver.go | 2 +-
 test/test-credentials.sh      | 8 ++++----
 test/test-extra-header.sh     | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/test/cmd/lfstest-gitserver.go b/test/cmd/lfstest-gitserver.go
index 87a1d4ee..97732995 100644
--- a/test/cmd/lfstest-gitserver.go
+++ b/test/cmd/lfstest-gitserver.go
@@ -1246,7 +1246,7 @@ func locksHandler(w http.ResponseWriter, r *http.Request, repo string) {
 }
 
 func missingRequiredCreds(w http.ResponseWriter, r *http.Request, repo string) bool {
-	if repo != "requirecreds" {
+	if !strings.HasPrefix(repo, "requirecreds") {
 		return false
 	}
 
diff --git a/test/test-credentials.sh b/test/test-credentials.sh
index 0ebf93a2..3efbe6b5 100755
--- a/test/test-credentials.sh
+++ b/test/test-credentials.sh
@@ -294,9 +294,9 @@ begin_test "credentials from lfs.url"
 (
   set -e
 
-  reponame="requirecreds"
+  reponame="requirecreds-lfsurl"
   setup_remote_repo "$reponame"
-  clone_repo "$reponame" requirecreds-lfsurl
+  clone_repo "$reponame" "$reponame"
 
   git lfs track "*.dat"
   echo "push a" > a.dat
@@ -335,9 +335,9 @@ begin_test "credentials from remote.origin.url"
 (
   set -e
 
-  reponame="requirecreds"
+  reponame="requirecreds-remoteurl"
   setup_remote_repo "$reponame"
-  clone_repo "$reponame" requirecreds-remoteurl
+  clone_repo "$reponame" "$reponame"
 
   git lfs track "*.dat"
   echo "push b" > b.dat
diff --git a/test/test-extra-header.sh b/test/test-extra-header.sh
index 823a7d62..1dbcfe30 100755
--- a/test/test-extra-header.sh
+++ b/test/test-extra-header.sh
@@ -30,7 +30,7 @@ begin_test "http.<url>.extraHeader with authorization"
 (
   set -e
 
-  reponame="requirecreds"
+  reponame="requirecreds-extraHeader"
   setup_remote_repo "$reponame"
   clone_repo "$reponame" "$reponame"
 
-- 
2.14.3

